scipy/stats/_multivariate.py

```diff
@@ -2648,6 +2661,394 @@ def entropy(self):
+    `p` parameters, returning a "frozen" multinomial random variable:
+
+    The probability mass function for `multinomial` is
+
+    .. math::
+
+        f(x) = \begin{pmatrix} n \\ x_1, \ldots, x_k \end{pmatrix}
+        p_1^{x_1} \cdots p_k^{x_k},
+
+    supported on :math:`x=(x_1, \ldots, x_k)` where each :math:`x_i` is a
+    nonnegative integer and their sum is :math:`n`.
+
+    Examples
+    --------
+
+    >>> from scipy.stats import multinomial
+    >>> rv = multinomial(8, [0.3, 0.2, 0.5])
+    >>> rv.pmf([1, 3, 4])
+    0.042000000000000072
+
+    The functions `pmf`, `logpmf`, `entropy`, and `cov` support broadcasting,
+    under the convention that the vector parameters (`x` and `p`) are
+    interpreted as if each row along the last axis is a single object. For
+    instance:
+
+    >>> multinomial.pmf([[3, 4], [3, 5]], [7, 8], [.3, .7])
+    array([0.2268945,  0.25412184])
+
+    Here, `x.shape == (2, 2)`, `n.shape == (2,)`, and `p.shape == (2,)`, but
+    following the rules mentioned above they follow numpy's broadcasting rules
+    as if the rows `[3, 4]` and `[3, 5]` in `x` and `[.3, .7]` in `p` were a
+    single object, and as if we had `x.shape = (2,)`, `n.shape = (2,)`, and
+    `p.shape = (,)`.
+
+    This broadcasting also works for `cov`, where the output objects
+    are square matrices of size `p.shape[-1]`. For example:
+
+    >>> multinomial.cov([4, 5], [[.3, .7], [.4, .6]])
+    array([[[ 0.84, -0.84],
+            [-0.84,  0.84]],
+           [[ 1.2 , -1.2 ],
+            [-1.2 ,  1.2 ]]])
+
+    In this example, `n.shape == (2,)` and `p.shape == (2, 2)`, and
+    following the rules above, these broadcast as if
+    `p.shape == (2,)`. Thus the result should also be of shape
+    `(2,), but since each output is a :math:`2 \times 2` matrix, the result
+    in fact has shape `(2, 2, 2)`.
+
+    Currently, `rvs` does not support broadcasting. Instead, the first element
+    of `n` and the first vector in `p` are used to choose the distribution from
+    which to sample. The current behavior should not be relied on.
+    """
+
+    def __init__(self, seed=None):
+        super(multinomial_gen, self).__init__(seed)
+        self.__doc__ = \
+            doccer.docformat(self.__doc__, multinomial_docdict_params)
+
+    def __call__(self, n, p, seed=None):
+        """
+        Create a frozen multinomial distribution.
+
+        See `multinomial_frozen` for more information.
+        """
+        return multinomial_frozen(n, p, seed)
+
+    def _process_parameters(self, n, p):
+        p = np.array(p, dtype=np.float64, copy=True)
+        p[...,-1] = 1. - p[...,:-1].sum(axis=-1)
+
+        if np.any(p <= 0) or np.any(p > 1):
+            raise ValueError("Elements of p should be between 0 and 1 "
+                    "and sum to 1.")
+
+        n = np.array(n, dtype=np.int, copy=True)
+
+        if np.any(n <= 0):
+            raise ValueError("Must have n > 0.")
```

Comment:
This looks reasonable but not exactly consistent with what other parts of scipy are doing (e.g. univariate distributions, special functions etc). I think a common practice is to fill invalid parts of the input with nans. That is, if `n = [8, -1]` the current version is going to raise.  However,

```
In [8]: from scipy.stats import binom

In [9]: binom.pmf(1, [2, -1], 0.1)
Out[9]: array([ 0.18,   nan])
```